backend/FwLite/MiniLcm/Validators/SenseValidator.cs

```diff
@@ -21,4 +22,19 @@ public SenseValidator(Entry entry): this()
         //it's ok if senses EntryId is an Empty guid
         RuleFor(s => s.EntryId).Equal(entry.Id).When(s => s.EntryId != Guid.Empty).WithMessage(sense => $"Sense (Id: {sense.Id}) EntryId must match Entry {entry.Id}, but instead was {sense.EntryId}");
     }
+
+    private string GetSenseIdentifier(Sense sense)
+    {
+        var firstGloss = sense.Gloss?.Values?.Values.FirstOrDefault();
+        if (string.IsNullOrWhiteSpace(firstGloss))
+        {
+            return sense.Id.ToString("D");
+        }
+        return $"{sense.Gloss} - {sense.Id}";
+    }
+
+    protected override void RaiseValidationException(ValidationContext<Sense> context, ValidationResult result)
+    {
+        throw new ValidationException(GetSenseIdentifier(context.InstanceToValidate), result.Errors, true);
```

Comment:
I suggest overriding this validation error for context. Though this does mean that context won't get added when validating `Entry.Sense` so we may need to handle that a different way, but this way we don't have to specify the context for each property.